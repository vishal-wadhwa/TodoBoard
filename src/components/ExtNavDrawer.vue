<template>
  <v-navigation-drawer
    :value='state'
    @input='val => $emit("end:change", val)'
    fixed
    clipped
    app
    class='pa-2'
    width='280'
  >
    <v-layout column>
      <v-flex class='text-xs-center'>
        <v-avatar size='200'>
          <img src='https://picsum.photos/300/?random' class='img-avatar--border'>
        </v-avatar>
      </v-flex>
      <v-list :subheader='true' dense>
        <v-list-tile>
          <v-subheader class='mt-3'>Boards</v-subheader>
        </v-list-tile>

        <v-list-tile
          avatar
          v-for='board in boardList'
          :key='board._id'
          :to='{name: "home", params: {"boardId": board._id}}'
          @click='ev => $emit("end:board-click", board, ev)'
        >
          <v-list-tile-avatar>
            <v-icon>chrome_reader_mode</v-icon>
          </v-list-tile-avatar>
          <v-list-tile-title v-text='board.boardName'></v-list-tile-title>
        </v-list-tile>
      </v-list>
      <v-text-field
        ref='bnameFieldBox'
        v-show='showNewFieldBox'
        clearable
        placeholder='My board'
        class='ml-4 my-0 pa-0 body-1'
        flat
        v-model='newBoardName'
        single-line
        @keydown.enter='onBoardNameSave'
        @keydown.esc='onBoardNameDiscard'
        @click:append-outer='onBoardNameSave'
        append-outer-icon='check_circle'
      ></v-text-field>
    </v-layout>
  </v-navigation-drawer>
</template>
<script>
export default {
  model: {
    prop: 'state',
    event: 'end:change'
  },
  props: {
    state: {
      type: Boolean,
      default: true
    },
    boardList: {
      type: Array,
      default: () => []
    },
    showNewFieldBox: {
      type: Boolean,
      default: false
    }
  },
  data () {
    return {
      newBoardName: ''
    }
  },
  methods: {
    onBoardNameSave (ev) {
      this.$emit('end:bname-save', this.newBoardName.trim(), ev)
      this.newBoardName = ''
    },
    onBoardNameDiscard (ev) {
      this.$emit('end:bname-discard', ev)
      this.newBoardName = ''
    }
  },
  watch: {
    showNewFieldBox (val) {
      if (val) {
        this.$nextTick(this.$refs.bnameFieldBox.focus)
      }
    }
  }
}
</script>
<style lang="stylus" scoped>
.img-avatar {
  &--border {
    border: 2px solid black;
  }
}
</style>
