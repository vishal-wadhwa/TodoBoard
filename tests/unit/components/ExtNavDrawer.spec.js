import { shallowMount, createLocalVue, mount } from '@vue/test-utils'
import ExtNavDrawer from '@/components/ExtNavDrawer'
import Vuetify from 'vuetify'
import Vue from 'vue'
import VueRouter from 'vue-router'

Vue.use(Vuetify)

describe('ExtNavDrawer.vue', () => {
  // required to vuetify [warn] in test
  // const app = document.createElement('div')
  // app.setAttribute('data-app', true)
  // document.body.appendChild(app)

  // suppresses vuetify internal TypeError, tests pass regardless
  // console.log = jest.fn()

  let localVue = null
  let router = null
  const factory = (opts = {}, full = false) => {
    return full ? mount(ExtNavDrawer, opts) : shallowMount(ExtNavDrawer, opts)
  }
  beforeEach(() => {
    localVue = createLocalVue()
    router = new VueRouter()
    localVue.use(router)
    // localVue.use(Vuetify)
  })

  const boardList = [
    { _id: 1, boardName: 'bjkgh1' },
    { _id: 2, boardName: 'bjhv2' },
    { _id: 3, boardName: 'bjhvkjh3' }
  ]
  const newBoardName = 'my-board-1234'

  it('renders the board names with titles correctly', () => {
    const wrapper = factory({
      propsData: { boardList },
      stubs: ['v-list-tile-title'],
      localVue
    })

    const listItems = wrapper.findAll({ name: 'v-list-tile' })

    expect(listItems.length).toBe(boardList.length + 1)
    for (let i = 0; i < boardList.length; ++i) {
      expect(listItems.at(1 + i).text()).toMatch(boardList[i].boardName)
      expect(listItems.at(i + 1).attributes('title')).toBe(boardList[i].boardName)
    }
  })

  it('deletes the key of the board on which delete is triggered', async () => {
    const wrapper = factory({
      propsData: { boardList },
      stubs: ['router-link'],
      localVue,
      sync: false
    }, true)

    const delButton = wrapper.find({ name: 'v-btn' })
    const listItem = wrapper.find('.v-list__tile[title]')

    expect(wrapper.vm.deleteBtnVisible[boardList[0]._id]).toBeUndefined()

    listItem.trigger('mouseover')
    await wrapper.vm.$nextTick()

    expect(wrapper.vm.deleteBtnVisible[boardList[0]._id]).toBeTruthy()

    delButton.trigger('click')
    await wrapper.vm.$nextTick()

    expect(wrapper.vm.deleteBtnVisible[boardList[0]._id]).toBeUndefined()
  })

  it('mouseover/mouseout toggle visibility of deleteButton', async () => {
    const wrapper = factory({
      propsData: { boardList },
      stubs: ['router-link'],
      localVue,
      sync: false
    }, true)

    const delButton = wrapper.find({ name: 'v-btn' })
    // const listItem = wrapper.findAll({ name: 'v-list-tile' }).at(1)
    const listItem = wrapper.find('.v-list__tile[title]')

    expect(delButton.isVisible()).toBeFalsy()

    listItem.trigger('mouseover')
    await wrapper.vm.$nextTick()

    expect(delButton.isVisible()).toBeTruthy()

    listItem.trigger('mouseout')
    await wrapper.vm.$nextTick()

    expect(delButton.isVisible()).toBeFalsy()
  })

  it('updates the list view when a new board is added', async () => {
    const boardItem = { _id: 4, boardName: '___34rfa45#$$@__aefq4' }
    const boardListCopy = [...boardList]

    const wrapper = factory({
      propsData: { boardList: boardListCopy },
      localVue,
      stubs: ['router-link'],
      sync: false
    }, true)

    const list = wrapper.find({ name: 'v-list' })
    expect(list.text()).not.toMatch(boardItem.boardName)

    boardListCopy.push(boardItem)

    await wrapper.vm.$nextTick()

    expect(wrapper.props('boardList')[3]).toEqual(boardItem)
    expect(list.findAll({ name: 'v-list-tile' }).length).toBe(boardListCopy.length + 1)
    expect(list.text()).toMatch(boardItem.boardName)
  })

  it('updates the list view when a board is deleted', async () => {
    const boardListCopy = [...boardList]

    const wrapper = factory({
      propsData: { boardList: boardListCopy },
      stubs: ['router-link'],
      localVue,
      sync: false
    }, true)

    const list = wrapper.find({ name: 'v-list' })
    const boardNameToDelete = boardListCopy[0].boardName

    expect(list.text()).toMatch(boardNameToDelete)

    boardListCopy.splice(0, 1)

    await wrapper.vm.$nextTick()

    expect(wrapper.props('boardList')[0].boardName).not.toMatch(boardNameToDelete)
    expect(list.text()).not.toMatch(boardNameToDelete)
  })

  it('displays text-field for new board only after it is toggled', async () => {
    const wrapper = factory({
      localVue,
      sync: false
    }, true)

    expect(wrapper.find({ name: 'v-text-field' }).isVisible()).toBeFalsy()
    wrapper.setProps({ showNewFieldBox: true })

    await wrapper.vm.$nextTick()
    expect(wrapper.find({ name: 'v-text-field' }).isVisible()).toBeTruthy()
  })

  it('has appropriate "to" property attached to list items', () => {
    const wrapper = factory({
      propsData: { boardList },
      stubs: ['router-link'],
      localVue,
      sync: false
    }, true)

    const routeLinks = wrapper.findAll({ name: 'v-list-tile' })

    for (let i = 1; i < routeLinks.length; ++i) {
      expect(routeLinks.at(i).props().to).toEqual(
        {
          name: 'home',
          params: { boardId: boardList[i - 1]._id }
        })
    }
  })

  it('emits "end:board-click" when a list item is clicked', async () => {
    const wrapper = factory({
      propsData: { boardList },
      stubs: ['router-link'],
      localVue,
      sync: false
    }, true)

    const listItems = wrapper.findAll('.v-list__tile[title]')
    listItems.at(0).trigger('click')
    listItems.at(2).trigger('click')

    await wrapper.vm.$nextTick()

    const emitArr = wrapper.emitted('end:board-click')

    expect(emitArr.length).toBe(2)

    expect(emitArr[0].length).toBe(2)
    expect(emitArr[0][0]).toEqual(boardList[0]._id)
    expect(emitArr[0][1]).toBeInstanceOf(Event)

    expect(emitArr[1].length).toBe(2)
    expect(emitArr[1][0]).toEqual(boardList[2]._id)
    expect(emitArr[1][1]).toBeInstanceOf(Event)
  })

  it('emits "end:board-delete" when delete button is clicked', async () => {
    const wrapper = factory({
      propsData: { boardList },
      stubs: ['router-link'],
      localVue,
      sync: false
    }, true)

    const delButtons = wrapper.findAll('.v-btn')
    delButtons.at(0).trigger('click')
    delButtons.at(2).trigger('click')

    await wrapper.vm.$nextTick()

    const emitArr = wrapper.emitted('end:board-delete')

    expect(emitArr.length).toBe(2)

    expect(emitArr[0].length).toBe(2)
    expect(emitArr[0][0]).toEqual(boardList[0]._id)
    expect(emitArr[0][1]).toBeInstanceOf(Event)

    expect(emitArr[1].length).toBe(2)
    expect(emitArr[1][0]).toEqual(boardList[2]._id)
    expect(emitArr[1][1]).toBeInstanceOf(Event)
  })

  it('emits "end:bname-save" when enter-key is pressed', async () => {
    const wrapper = factory({
      localVue,
      propsData: { showNewFieldBox: true },
      sync: false
    }, true)

    wrapper.setData({ newBoardName })

    const textField = wrapper.find('input')
    textField.trigger('keydown.enter')

    await wrapper.vm.$nextTick()

    const emitArr = wrapper.emitted('end:bname-save')
    expect(emitArr.length).toBe(1)
    expect(emitArr[0].length).toBe(2)
    expect(emitArr[0][0]).toBe(newBoardName)
    expect(emitArr[0][1]).toBeInstanceOf(Event)
  })

  it('emits "end:bname-save" when tick icon is clicked', async () => {
    const wrapper = factory({
      localVue,
      propsData: { showNewFieldBox: true },
      sync: false
    }, true)

    wrapper.setData({ newBoardName })

    const icon = wrapper.find('input')
    icon.trigger('click:append-outer')

    await wrapper.vm.$nextTick()

    const emitArr = wrapper.emitted('end:bname-save')
    expect(emitArr.length).toBe(1)
    expect(emitArr[0].length).toBe(2)
    expect(emitArr[0][0]).toBe(newBoardName)
    expect(emitArr[0][1]).toBeInstanceOf(Event)
  })

  it('emits "end:bname-discard" when esc-key is pressed', async () => {
    const wrapper = factory({
      localVue,
      propsData: { showNewFieldBox: true },
      sync: false
    }, true)

    wrapper.setData({ newBoardName })

    const textField = wrapper.find('input')
    textField.trigger('keydown.esc')

    await wrapper.vm.$nextTick()

    const emitArr = wrapper.emitted('end:bname-discard')
    expect(emitArr.length).toBe(1)
    expect(emitArr[0].length).toBe(1)
    expect(emitArr[0][0]).toBeInstanceOf(Event)
  })

  it('emits "end:blur" on textField blur', async () => {
    const wrapper = factory({
      localVue,
      propsData: { showNewFieldBox: true },
      sync: false
    }, true)

    const textField = wrapper.find('input')
    textField.trigger('blur')

    await wrapper.vm.$nextTick()

    const emitArr = wrapper.emitted('end:blur')
    expect(emitArr.length).toBe(1)
    expect(emitArr[0].length).toBe(1)
    expect(emitArr[0][0]).toBeInstanceOf(Event)
  })

  it('does not emit "end:bname-save" and goes into error state when boardName is empty', async () => {
    const wrapper = factory({
      localVue,
      propsData: { showNewFieldBox: true },
      sync: false
    }, true)

    const textField = wrapper.find('input')

    expect(wrapper.vm.errMsg).toBe('')

    textField.trigger('keydown.enter')

    await wrapper.vm.$nextTick()

    const emitArr = wrapper.emitted('end:bname-save')
    expect(emitArr).toBeUndefined()
    expect(wrapper.vm.errMsg).toBe(wrapper.vm.ERROR_MSG)
  })

  it('should match snapshot when text field is not visible', () => {
    const wrapper = factory({
      localVue,
      propsData: { boardList },
      stubs: ['router-link'],
      sync: false
    }, true)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot when text field is visible', () => {
    const wrapper = factory({
      localVue,
      propsData: { boardList, showNewFieldBox: true },
      stubs: ['router-link'],
      sync: false
    }, true)
    wrapper.setData({ newBoardName })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot when delete button is visible', async () => {
    const wrapper = factory({
      localVue,
      propsData: { boardList },
      stubs: ['router-link'],
      sync: false
    }, true)
    wrapper.setData({ newBoardName })

    const listItem = wrapper.find('.v-list__tile[title]')
    listItem.trigger('mouseover')

    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })
})
