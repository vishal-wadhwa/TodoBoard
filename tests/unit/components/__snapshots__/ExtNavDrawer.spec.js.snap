// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtNavDrawer.vue should match snapshot when delete button is visible 1`] = `
<aside class="py-2 v-navigation-drawer v-navigation-drawer--clipped v-navigation-drawer--fixed v-navigation-drawer--is-mobile v-navigation-drawer--open theme--light" style="height: 100%; margin-top: 0px; max-height: calc(100% - 0px); transform: translateX(0px); width: 280px;">
  <div class="layout column">
    <div class="flex text-xs-center">
      <div class="v-avatar" style="height: 200px; width: 200px;"><img src="https://picsum.photos/300/?random" class="img-avatar--border"></div>
    </div>
    <div class="v-list v-list--dense v-list--subheader theme--light">
      <div>
        <div class="v-list__tile theme--light">
          <div class="v-subheader mt-3 theme--light">Boards</div>
        </div>
      </div>
      <div>
        <router-link-stub title="bjkgh1" class="v-list__tile v-list__tile--link v-list__tile--avatar theme--light">
          <div class="v-list__tile__avatar">
            <div class="v-avatar" style="height: 40px; width: 40px;"><i aria-hidden="true" class="v-icon material-icons theme--light">chrome_reader_mode</i></div>
          </div>
          <div class="v-list__tile__title">bjkgh1</div> <button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--small theme--light" title="" style="">
            <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">delete</i></div>
          </button>
        </router-link-stub>
      </div>
      <div>
        <router-link-stub title="bjhv2" class="v-list__tile v-list__tile--link v-list__tile--avatar theme--light">
          <div class="v-list__tile__avatar">
            <div class="v-avatar" style="height: 40px; width: 40px;"><i aria-hidden="true" class="v-icon material-icons theme--light">chrome_reader_mode</i></div>
          </div>
          <div class="v-list__tile__title">bjhv2</div> <button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--small theme--light" title="" style="display: none;">
            <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">delete</i></div>
          </button>
        </router-link-stub>
      </div>
      <div>
        <router-link-stub title="bjhvkjh3" class="v-list__tile v-list__tile--link v-list__tile--avatar theme--light">
          <div class="v-list__tile__avatar">
            <div class="v-avatar" style="height: 40px; width: 40px;"><i aria-hidden="true" class="v-icon material-icons theme--light">chrome_reader_mode</i></div>
          </div>
          <div class="v-list__tile__title">bjhvkjh3</div> <button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--small theme--light" title="" style="display: none;">
            <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">delete</i></div>
          </button>
        </router-link-stub>
      </div>
    </div>
    <div class="v-input ml-4 my-0 pa-0 body-1 v-text-field v-text-field--single-line v-text-field--solo-flat v-text-field--placeholder v-input--is-label-active v-input--is-dirty theme--light" style="display: none;">
      <div class="v-input__control">
        <div class="v-input__slot">
          <div class="v-text-field__slot"><input type="text" placeholder="My board"></div>
          <div class="v-input__append-inner">
            <div class="v-input__icon v-input__icon--clear"><i aria-hidden="true" class="v-icon v-icon--link material-icons theme--light">clear</i></div>
          </div>
        </div>
        <div class="v-text-field__details">
          <div class="v-messages theme--light">
            <div class="v-messages__wrapper"></div>
          </div>
        </div>
      </div>
      <div class="v-input__append-outer">
        <div class="v-input__icon v-input__icon--append-outer"><i aria-hidden="true" class="v-icon v-icon--link material-icons theme--light">check_circle</i></div>
      </div>
    </div>
  </div>
  <div class="v-navigation-drawer__border"></div>
</aside>
`;

exports[`ExtNavDrawer.vue should match snapshot when text field is not visible 1`] = `
<aside class="py-2 v-navigation-drawer v-navigation-drawer--clipped v-navigation-drawer--fixed v-navigation-drawer--is-mobile v-navigation-drawer--open theme--light" style="height: 100%; margin-top: 0px; max-height: calc(100% - 0px); transform: translateX(0px); width: 280px;">
  <div class="layout column">
    <div class="flex text-xs-center">
      <div class="v-avatar" style="height: 200px; width: 200px;"><img src="https://picsum.photos/300/?random" class="img-avatar--border"></div>
    </div>
    <div class="v-list v-list--dense v-list--subheader theme--light">
      <div>
        <div class="v-list__tile theme--light">
          <div class="v-subheader mt-3 theme--light">Boards</div>
        </div>
      </div>
      <div>
        <router-link-stub title="bjkgh1" class="v-list__tile v-list__tile--link v-list__tile--avatar theme--light">
          <div class="v-list__tile__avatar">
            <div class="v-avatar" style="height: 40px; width: 40px;"><i aria-hidden="true" class="v-icon material-icons theme--light">chrome_reader_mode</i></div>
          </div>
          <div class="v-list__tile__title">bjkgh1</div> <button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--small theme--light" title="" style="display: none;">
            <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">delete</i></div>
          </button>
        </router-link-stub>
      </div>
      <div>
        <router-link-stub title="bjhv2" class="v-list__tile v-list__tile--link v-list__tile--avatar theme--light">
          <div class="v-list__tile__avatar">
            <div class="v-avatar" style="height: 40px; width: 40px;"><i aria-hidden="true" class="v-icon material-icons theme--light">chrome_reader_mode</i></div>
          </div>
          <div class="v-list__tile__title">bjhv2</div> <button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--small theme--light" title="" style="display: none;">
            <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">delete</i></div>
          </button>
        </router-link-stub>
      </div>
      <div>
        <router-link-stub title="bjhvkjh3" class="v-list__tile v-list__tile--link v-list__tile--avatar theme--light">
          <div class="v-list__tile__avatar">
            <div class="v-avatar" style="height: 40px; width: 40px;"><i aria-hidden="true" class="v-icon material-icons theme--light">chrome_reader_mode</i></div>
          </div>
          <div class="v-list__tile__title">bjhvkjh3</div> <button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--small theme--light" title="" style="display: none;">
            <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">delete</i></div>
          </button>
        </router-link-stub>
      </div>
    </div>
    <div class="v-input ml-4 my-0 pa-0 body-1 v-text-field v-text-field--single-line v-text-field--solo-flat v-text-field--placeholder theme--light" style="display: none;">
      <div class="v-input__control">
        <div class="v-input__slot">
          <div class="v-text-field__slot"><input type="text" placeholder="My board"></div>
          <div class="v-input__append-inner">
            <div class="v-input__icon v-input__icon--"><i aria-hidden="true" class="v-icon v-icon--link material-icons theme--light"></i></div>
          </div>
        </div>
        <div class="v-text-field__details">
          <div class="v-messages theme--light">
            <div class="v-messages__wrapper"></div>
          </div>
        </div>
      </div>
      <div class="v-input__append-outer">
        <div class="v-input__icon v-input__icon--append-outer"><i aria-hidden="true" class="v-icon v-icon--link material-icons theme--light">check_circle</i></div>
      </div>
    </div>
  </div>
  <div class="v-navigation-drawer__border"></div>
</aside>
`;

exports[`ExtNavDrawer.vue should match snapshot when text field is visible 1`] = `
<aside class="py-2 v-navigation-drawer v-navigation-drawer--clipped v-navigation-drawer--fixed v-navigation-drawer--is-mobile v-navigation-drawer--open theme--light" style="height: 100%; margin-top: 0px; max-height: calc(100% - 0px); transform: translateX(0px); width: 280px;">
  <div class="layout column">
    <div class="flex text-xs-center">
      <div class="v-avatar" style="height: 200px; width: 200px;"><img src="https://picsum.photos/300/?random" class="img-avatar--border"></div>
    </div>
    <div class="v-list v-list--dense v-list--subheader theme--light">
      <div>
        <div class="v-list__tile theme--light">
          <div class="v-subheader mt-3 theme--light">Boards</div>
        </div>
      </div>
      <div>
        <router-link-stub title="bjkgh1" class="v-list__tile v-list__tile--link v-list__tile--avatar theme--light">
          <div class="v-list__tile__avatar">
            <div class="v-avatar" style="height: 40px; width: 40px;"><i aria-hidden="true" class="v-icon material-icons theme--light">chrome_reader_mode</i></div>
          </div>
          <div class="v-list__tile__title">bjkgh1</div> <button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--small theme--light" title="" style="display: none;">
            <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">delete</i></div>
          </button>
        </router-link-stub>
      </div>
      <div>
        <router-link-stub title="bjhv2" class="v-list__tile v-list__tile--link v-list__tile--avatar theme--light">
          <div class="v-list__tile__avatar">
            <div class="v-avatar" style="height: 40px; width: 40px;"><i aria-hidden="true" class="v-icon material-icons theme--light">chrome_reader_mode</i></div>
          </div>
          <div class="v-list__tile__title">bjhv2</div> <button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--small theme--light" title="" style="display: none;">
            <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">delete</i></div>
          </button>
        </router-link-stub>
      </div>
      <div>
        <router-link-stub title="bjhvkjh3" class="v-list__tile v-list__tile--link v-list__tile--avatar theme--light">
          <div class="v-list__tile__avatar">
            <div class="v-avatar" style="height: 40px; width: 40px;"><i aria-hidden="true" class="v-icon material-icons theme--light">chrome_reader_mode</i></div>
          </div>
          <div class="v-list__tile__title">bjhvkjh3</div> <button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--small theme--light" title="" style="display: none;">
            <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">delete</i></div>
          </button>
        </router-link-stub>
      </div>
    </div>
    <div class="v-input ml-4 my-0 pa-0 body-1 v-text-field v-text-field--single-line v-text-field--solo-flat v-text-field--placeholder theme--light">
      <div class="v-input__control">
        <div class="v-input__slot">
          <div class="v-text-field__slot"><input type="text" placeholder="My board"></div>
          <div class="v-input__append-inner">
            <div class="v-input__icon v-input__icon--"><i aria-hidden="true" class="v-icon v-icon--link material-icons theme--light"></i></div>
          </div>
        </div>
        <div class="v-text-field__details">
          <div class="v-messages theme--light">
            <div class="v-messages__wrapper"></div>
          </div>
        </div>
      </div>
      <div class="v-input__append-outer">
        <div class="v-input__icon v-input__icon--append-outer"><i aria-hidden="true" class="v-icon v-icon--link material-icons theme--light">check_circle</i></div>
      </div>
    </div>
  </div>
  <div class="v-navigation-drawer__border"></div>
</aside>
`;
